projects/core/src/site-theme/facade/site-theme.service.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+
+import { select, Store } from '@ngrx/store';
+import { filter, map, mergeMap, take, tap } from 'rxjs/operators';
+
+import { Observable } from 'rxjs';
+
+import { SiteTheme } from '../../model/misc.model';
+import { SiteThemeConfig } from '../config/site-theme-config';
+import { isNotNullable } from '../../util/type-guards';
+import { StateWithSiteTheme } from '../store/state';
+import { SiteThemeSelectors } from '../store/selectors';
+import { SiteThemeActions } from '../store/actions';
+
+@Injectable()
+export class SiteThemeService {
+  private _isInitialized = false;
+  constructor(
+    protected store: Store<StateWithSiteTheme>,
+    protected config: SiteThemeConfig
+  ) {}
+
+  getAll(): Observable<SiteTheme[]> {
+    return this.store.pipe(
+      select(SiteThemeSelectors.getAllSiteThemes),
+      tap((themes) => {
+        if (!themes) {
+          this.store.dispatch(new SiteThemeActions.LoadSiteThemes());
+        }
+      }),
+      filter(isNotNullable)
+    );
+  }
+
+  /**
+   * Gets the className of the active theme.
+   */
+  getActive(): Observable<string> {
+    return this.store.pipe(
+      select(SiteThemeSelectors.getActiveSiteTheme),
+      filter(isNotNullable)
+    );
+  }
+
+  /**
+   * Sets the active theme className.
+   */
+  setActive(className: string): void {
+    this.isValidTheme(className)
+      .pipe(
+        filter((isValid) => isValid),
+        mergeMap(() => {
+          return this.store.pipe(
+            select(SiteThemeSelectors.getActiveSiteTheme),
+            take(1)
+          );
+        })
+      )
+      .subscribe((activeTheme) => {
+        if (activeTheme !== className) {
+          this.store.dispatch(
+            new SiteThemeActions.SetActiveSiteTheme(className)
+          );
+        }
+      });
+    this._isInitialized = true;
+  }
```

Comment:
thanks, I will wrap the subscription in setActive with a Subscription, but I will keep _isInitialized separate because it can be called from two places, and the second call might end up being uninitialized due to waiting for the subscription. We only need _isInitialized to check if the theme has been initialized.

Reply:
moved  _isInitialized inside